app/controllers/RestaurantController.java

```diff
@@ -114,8 +114,13 @@ public Result getNearbyRestaurants(Float latitude, Float longitude) {
 	 * @return the popular restaurants
 	 */
 	@Transactional(readOnly = true)
-	public Result getPopularRestaurants() {
-		return wrapForPublic(() -> this.service.getPopularRestaurants());
+	public Result getPopularRestaurants(String id) {
+	        if(id != "undefined"){
```

Comment:
There must not be any logic outside the wrapper in Controllers. You could pass the ID as a string and try to convert it to a UUID in the service method.